mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -160,8 +165,8 @@ class LogisticRegression(override val uid: String)
       new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, regParamL1Fun, $(tol))
     }
 
-    val initialWeightsWithIntercept =
-      Vectors.zeros(if ($(fitIntercept)) numFeatures + 1 else numFeatures)
+    val initialWeightsWithIntercept = optInitialWeights.getOrElse(
+      Vectors.zeros(if ($(fitIntercept)) numFeatures + 1 else numFeatures))
 
```

Comment:
If we call `ml` version from `mllib` side by creating `DataFrame` in `LogisticRegressionWithLBFGS`, we can handle the model format conversion in `mllib`. In new `ml` package, the multinomial LoR will have models as matrix, so if we change too much here, will be difficult to implement MLoR later in `ml`.